plugins/Marketplace/PackageList.py

```diff
@@ -91,3 +126,214 @@ def hasFooter(self) -> bool:
+            # between de-/constructing Remote or Local PackageLists. This try-except is here to prevent a hard crash when the wrapped C++ object
+            # was deleted when it was still parsing the response
+            return
+
+    def _downloadError(self, package_id: str, update: bool = False, reply: Optional["QNetworkReply"] = None, error: Optional["QNetworkReply.NetworkError"] = None) -> None:
+        if reply:
+            reply_string = bytes(reply.readAll()).decode()
+            Logger.error(f"Failed to download package: {package_id} due to {reply_string}")
+        try:
+            package = self.getPackageModel(package_id)
+            if update:
+                package.is_updating = ManageState.FAILED
+            else:
+                package.is_installing = ManageState.FAILED
+        except RuntimeError:
+            # Setting the ownership of this object to not qml can still result in a RuntimeError. Which can occur when quickly toggling
+            # between de-/constructing Remote or Local PackageLists. This try-except is here to prevent a hard crash when the wrapped C++ object
+            # was deleted when it was still parsing the response
+            return
+
+    def subscribeUserToPackage(self, package_id: str, sdk_version: str) -> None:
+        """Subscribe the user (if logged in) to the package for a given SDK
+
+         :param package_id: the package identification string
+         :param sdk_version: the SDK version
+         """
+        if self._account.isLoggedIn:
+            HttpRequestManager.getInstance().put(
+                url = USER_PACKAGES_URL,
+                data = json.dumps({"data": {"package_id": package_id, "sdk_version": sdk_version}}).encode(),
+                scope = self._scope
+            )
+
+    def unsunscribeUserFromPackage(self, package_id: str) -> None:
+        """Unsubscribe the user (if logged in) from the package
+
+         :param package_id: the package identification string
+         """
+        if self._account.isLoggedIn:
+            HttpRequestManager.getInstance().delete(url = f"{USER_PACKAGES_URL}/{package_id}", scope = self._scope)
+
+    # --- Handle the manage package buttons ---
+
+    def _connectManageButtonSignals(self, package: PackageModel) -> None:
+        package.installPackageTriggered.connect(self.installPackage)
+        package.uninstallPackageTriggered.connect(self.uninstallPackage)
+        package.updatePackageTriggered.connect(self.updatePackage)
+        package.enablePackageTriggered.connect(self.enablePackage)
+        package.disablePackageTriggered.connect(self.disablePackage)
+
+    @pyqtSlot(str)
+    def installPackage(self, package_id: str) -> None:
+        """Install a package from the Marketplace
+
+        :param package_id: the package identification string
+        """
+        package = self.getPackageModel(package_id)
+        package.is_installing = ManageState.PROCESSING
```

Comment:
Why not let the package set that itself? It's already emitting the signal, right?